proxy/proxy.go

```diff
@@ -501,6 +501,10 @@ func mapRequest(ctx *context, requestContext stdlibcontext.Context, removeHopHea
 	} else {
 		rr.Header = cloneHeader(r.Header)
 	}
+	// Disable default net/http user agent when user agent is not specified
+	if _, ok := rr.Header["User-Agent"]; !ok {
+		rr.Header["User-Agent"] = []string{""}
```

Comment:
I tested that this works, too and reads better:
```diff
diff --git proxy/proxy.go proxy/proxy.go
index e04f510d..91d5d5be 100644
--- proxy/proxy.go
+++ proxy/proxy.go
@@ -501,6 +501,10 @@ func mapRequest(ctx *context, requestContext stdlibcontext.Context, removeHopHea
        } else {
                rr.Header = cloneHeader(r.Header)
        }
+
+       if rr.Header.Get("User-Agent") == "" {
+               rr.Header.Set("User-Agent", "")
+       }
```

Reply:
I've implemented it this way because the fix for https://github.com/golang/go/issues/15524 as well as [request writer](https://cs.opensource.google/go/go/+/refs/tags/go1.18.1:src/net/http/request.go;l=625-630) both check for the presence of the value treating headers as a map.
The `rr.Header["User-Agent"]` explicitly checks for absent value while `rr.Header.Get("User-Agent")` returns empty for both absent and empty stored value, i.e. ambiguous. 
Also both `Set` and `Get` transform header name into canonical form which is redundant since `User-Agent` is already a canonical form (not a big deal probably but why spend cycles on the hot path if we know it is useless).
We also treat Header type as a map in the `copyHeader` so it should not be a concern.

> and reads better

The fix uses undocumented feature of request writer that does not use default user agent (also undocumented) if the value is explicitly set to empty. Without knowing this your version reads weird anyway - "set to empty if it is empty"...